frontend/src/features/admin-form/create/payment/PaymentDrawer.tsx

```diff
@@ -330,11 +340,16 @@ export const PaymentDrawer = ({
   }, [paymentsField, resetData, setData])
 
   // Allows for payment data refresh in encrypt mode
-  if (!paymentData && isEncryptMode) return null
+  if (!paymentData && isEncryptMode && isStripeConnected) return null
+
+  const isDisabled = !(isEncryptMode && isStripeConnected) || isLoading
 
   return (
     <CreatePageDrawerContainer>
-      <PaymentDisabledMessage isEmailMode={!isEncryptMode} />
+      <PaymentDisabledMessage
+        isEmailMode={!isEncryptMode}
+        isStripeUnconnected={!isStripeConnected}
```

Comment:
would recommend to use the "positive" versions rather than negating, for readability.